airbyte-lib/airbyte_lib/types.py

```diff
@@ -0,0 +1,80 @@
+"""Type conversion methods for SQL Caches."""
```

Comment:
This is a generic implementation and will need work as we test in diverse scenarios. The abstractions, I think, are solid though - so ideally we will just need to incrementally add handling when we see failures on edge cases.

Still TBD if we will need to override this for different SQL dialects, but so far, it seems to work nicely generic. If we do need to override, the Cache class already has a means to take an overriden subclass of this main implementation, and those overriden implementations could have special handling if we find that some cases can't be implemented generically.